liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -959,4 +854,102 @@ public void info(String msg) {
 
     }
 
+    private Plugin getLibertyPluging() {
+        Plugin plugin = project.getPlugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID + ":" +LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
+        if (plugin == null) {
+            plugin = plugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID,  "LATEST");
+        }
+        return plugin;
+    }
+
+    private Xpp3Dom getLibertyPluginConfig() {
+        // get the Liberty plugin configuration from the pom and overrides looseApplication to true.
+        // ignores the plugin execution configuration just like CLI invocation (e.g. mvn liberty:create) ignoring execution configuration.
+        Plugin libertyPlugin = getLibertyPluging();
+        Xpp3Dom overrides = configuration(element(name("looseApplication"), "true"));
+        return getPluginConfig(libertyPlugin, overrides);
+    }
+
+    private Xpp3Dom getPluginConfig(Plugin plugin, Xpp3Dom overrides) {
+        Xpp3Dom config = (overrides != null) ? overrides : configuration();
+        if (plugin.getConfiguration() == null) {
+            return config;
+        } else {
+            return Xpp3Dom.mergeXpp3Dom(config, (Xpp3Dom)plugin.getConfiguration());
+        }
+    }
+
+    private final ArrayList<String> commonParams = new ArrayList<>(Arrays.asList(
+            "installDirectory", "runtimeArchive", "runtimeArtifact", "libertyRuntimeVersion",
+            "install", "licenseArtifact", "serverName", "userDirectory", "outputDirectory",
+            "runtimeInstallDirectory", "refresh", "skip"
+            ));
+    
+    private final ArrayList<String> commonServerParams = new ArrayList<>(Arrays.asList(
+            "serverXmlFile", "configDirectory", "bootstrapProperties", "bootstrapPropertiesFile",
+            "jvmOptions", "jvmOptionsFile", "serverEnvFile"));
+    private ArrayList<String> createParams = new ArrayList<>(Arrays.asList(
+            "template", "libertySettingsFolder", "noPassword"
+            ));
+    
+    private ArrayList<String> deployParams = new ArrayList<>(Arrays.asList(
+            "appsDirectory", "stripVersion", "deployPackages", "looseApplication", "timeout"
+            ));
+    
+    private ArrayList<String> installFeatureParams = new ArrayList<>(Arrays.asList("features"));    
+
+    private Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
+        for (int i=0; i<config.getChildCount(); i++) {
+            if (!goalParams.contains(config.getChild(i).getName().trim())) {
+                config.removeChild(i);
```

Comment:
Removing a child from `config` while iterating through its children seems like it would cause problems.  It would be better to operate on a copy of `config` and remove things from that copy, then return the copy.

Reply:
I wondered about that too. At the least, you could collect which children should be removed, and then call `removeChild` for each one outside of the loop that is iterating.

Reply:
your comment is well taken and I am trying to be more efficient.  I have added an immediate `return` after the recursive call which is a bug.
```
 private Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
        for (int i=0; i<config.getChildCount(); i++) {
            if (!goalParams.contains(config.getChild(i).getName().trim())) {
                config.removeChild(i);
                stripConfigElements(config, goalParams);
                return config;
            }
        }
        return config;
    }
```

Reply:
Using recursion here would inflate the call stack according to the size of the config object.  It works but adds unnecessary overhead, since the same result can be achieved with loops such as with Cheryl's suggestion.

Reply:
something like this...
```
    private Xpp3Dom stripConfigElements(Xpp3Dom config, ArrayList<String> goalParams) {
        List<Integer> removeChildren = new ArrayList<Integer>();
        for (int i=0; i<config.getChildCount(); i++) {
            if (!goalParams.contains(config.getChild(i).getName().trim())) {
                removeChildren.add(i);
            }
        }
        Collections.reverse(removeChildren);
        for (int child : removeChildren) {
            config.removeChild(child);
        }
        return config;
    }
```

Reply:
Yes, that seems fine